mapie/multi_label_classification.py

```diff
@@ -0,0 +1,746 @@
+            If None, then samples are equally weighted.
+            If some weights are null,
+            their corresponding observations are removed
+            before the fitting process and hence have no prediction sets.
+
+            By default None.
+
+        Returns
+        -------
+        MapieMultiLabelClassifier
+            The model itself.
+        """
+        if not self._check_partial_fit_first_call():
+            delattr(self, "risks")
+        return self.partial_fit(X, y, sample_weight)
+
+    def predict(
+        self,
+        X: ArrayLike,
+        method: Optional[str] = "crc",
+        alpha: Optional[Union[float, Iterable[float]]] = None,
+        delta: Optional[float] = None,
+        bound: Optional[Union[str, None]] = "wsr"
+    ) -> Union[NDArray, Tuple[NDArray, NDArray]]:
+        """
+        Prediction prediction sets on new samples based on target confidence
+        interval.
+        Prediction sets for a given ``alpha`` are deduced from the computed
+        risks.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Test data.
+        method : Optional[str], optional
+            Method to choose for prediction interval estimates.
+            Choose among:
+
+            - "crc", for Conformal Risk Control. See [1] fot
+            more details.
+
+            - "rcps", based on computation of the upper bound
+            of the risk. See [2] for more details.
+
+            By default "crc".
+
+        alpha : Optional[Union[float, Iterable[float]]]
+            Can be a float, a list of floats, or a ``ArrayLike`` of floats.
+            Between 0 and 1, represent the uncertainty of the confidence
+            interval.
+            Lower ``alpha`` produce larger (more conservative) prediction
+            sets.
+            ``alpha`` is the complement of the target coverage level.
+            By default ``None``.
+        delta : Optional[float]
+            Can be a float, or ``None``. If using method="rcps", then it
+            can not be set to ``None``.
+            Between 0 and 1, the level of certainty with which we compute
+            the Upper Confidence Bound of the average risk.
+            Lower ``delta`` produce larger (more conservative) prediction
+            sets.
+            By default ``None``.
+        bound : Optional[Union[str, None]]
+            Method used to compute the Upper Confience Bound of the
+            average risk. Only necessary with the RCPS method.
+            By default "wsr"
+
+        Returns
+        -------
+        Union[NDArray, Tuple[NDArray, NDArray]]
+
+        - NDArray of shape (n_samples,) if alpha is None.
+
+        - Tuple[NDArray, NDArray] of shapes
+        (n_samples, n_classes) and (n_samples, n_classes, n_alpha)
+        if alpha is not None.
+        """
+
+        self.method = method
+        self._check_method()
```

Comment:
should not it be a function of method ? self._check_method(method) and be placed before
or _check_method do both

def _check_method(self, method):
        if method not in self.valid_methods_:
            raise ValueError(
                "Invalid method. "
                "Allowed values are" + " or ".join(self.valid_methods_)
            )
else:
      self.method = method

Reply:
no I don't think so, done this way is the rest of mapie classification

Reply:
OK